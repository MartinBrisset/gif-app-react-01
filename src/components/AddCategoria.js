import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategoria = ({setCategorias}) => {
    
    const [inputValue, setInputValue] = useState('')

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value )
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        // console.log('enviado');
        
        if (inputValue.trim().length > 2) {
            setCategorias( categorias => [ inputValue, ...categorias ])
            setInputValue('')
        }
        
        
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <input 
                    type='text'
                    value={inputValue}
                    onChange={ handleInputChange }
                    />
            </form>
        </>
    )
}

AddCategoria.propTypes = {
    setCategorias: PropTypes.func.isRequired
}
