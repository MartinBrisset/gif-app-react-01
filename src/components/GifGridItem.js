import React from 'react'

export const GifGridItem = ({id,title,url}) => {


    return (
        <div className='card animate__animated animate__fadeIn animate__delay-0.8s'>
            <img src={url} alt={title} />
            <p> {title} </p>
        </div>
    )
}
