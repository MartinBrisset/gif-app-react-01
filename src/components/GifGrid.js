import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({categoria}) => {

    const { loading, data:imgs } = useFetchGifs(categoria)

    return (
        <>
            <h3 className='animate__animated animate__fadeIn animate__delay-0.8s'> {categoria} </h3>

            {loading && <p className='animate__animated animate__flash'>Cargando...</p>}

            <div className='card-grid'>
                
                    { imgs.map( img => {
                        return <GifGridItem
                        key={img.id}
                        {...img} 
                        />
                    })}

            </div>
        
        </>

    )
}
