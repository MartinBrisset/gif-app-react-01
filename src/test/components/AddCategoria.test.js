import React from "react"
import { shallow } from "enzyme"
import { AddCategoria } from "../../components/AddCategoria"



describe('Pruebas en componente <AddCategoria  />', () => {

    const setCategorias = jest.fn()

    let wrapper = shallow( <AddCategoria setCategorias={setCategorias} />)
    
    beforeEach( () => {
        jest.clearAllMocks()
        
        wrapper = shallow( <AddCategoria setCategorias={setCategorias} />)
    })
    
    test('debe de mostrarse correctamente', () => {
        

        expect( wrapper ).toMatchSnapshot()


    })

    test('Debe de cambiar la caja de texto', () => {
        
        const input = wrapper.find('input')
        const value = 'Hola Mundo'

        input.simulate('change', { 
            target: {
                value
            } 
        })

    })

    test('No debe de postear la informacion con submit', () => {
        
        wrapper.find('form').simulate('submit', {
            preventDefault(){

            } 
        })

        expect( setCategorias ).not.toHaveBeenCalled()

    })

    test('debe de llamar el setCategorias y limpiar la caja de texto', () => {
        
        const value = 'Hola Mundo'
        wrapper.find('input').simulate('change', {
            target: {
                value
            }
        })

        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategorias ).toHaveBeenCalled()
        expect( setCategorias ).toHaveBeenCalledWith(expect.any(Function))

        expect( wrapper.find('input').prop('value')).toBe('')


    })
    
    
    
    

})
