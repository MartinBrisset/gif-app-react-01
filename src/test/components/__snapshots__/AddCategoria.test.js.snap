// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <AddCategoria  /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
